docs/ibc/light-clients/client-state.md

```diff
@@ -1,3 +1,67 @@
 <!--
 order: 2
--->
\ No newline at end of file
+-->
+
+# Implementing the `ClientState` interface
+
+Learn how to implement the [`Client State`](https://github.com/cosmos/ibc-go/blob/v6.0.0-rc1/modules/core/exported/client.go#L40) interface.
+
+## `ClientType` method
+
+`ClientType` should return a unique string identifier of the light client. This will be used when generating a client identifier.
+The format is created as follows: `ClientTypes-{N}` where `{N}` is the unique global nonce associated with a specific client.
+
+## `GetLatestHeight` method
+
+`GetLatestHeight` should return the latest block height that the client state represents. .
+
+## `Validate` method
+
+`Validate` should validate every client state field and should return an error if any value is invalid. The light client
+implementor is in charge of determining which checks are required. See the [tendermint light client implementation](https://github.com/cosmos/ibc-go/blob/v6.0.0-rc1/modules/light-clients/07-tendermint/types/client_state.go#L101)
+as a reference.
+
+## `Status` method
+
+`Status` must return the status of the client. Only `Active` clients are allowed to process packets. All
+possible Status types can be found [here](https://github.com/cosmos/ibc-go/blob/v6.0.0-rc1/modules/core/exported/client.go#L26-L36).
+
+## `ZeroCustomFields` method
+
+`ZeroCustomFields` should return a copy of the light client with all client customizable fields with their zero value. It should not mutate the fields of the light client.
+This method is used when [scheduling upgrades](https://github.com/cosmos/ibc-go/blob/v6.0.0-rc1/modules/core/02-client/keeper/proposal.go#L89). Upgrades are used to upgrade chain specific fields such as chainID and the unbonding period.
```

Comment:
```suggestion
This method is used when [scheduling upgrades](https://github.com/cosmos/ibc-go/blob/v6.0.0-rc1/modules/core/02-client/keeper/proposal.go#L89). Upgrades are used to upgrade chain specific fields. In the tendermint case, this may be the chainID or the unbonding period.
```